distgo/docker/docker_push.go

```diff
@@ -86,23 +87,30 @@ func runSingleDockerPush(
 	dockerID distgo.DockerID,
 	productTaskOutputInfo distgo.ProductTaskOutputInfo,
 	dryRun bool,
+	insecure bool,
 	stdout io.Writer) (rErr error) {
 
 	// if an OCI artifact exists, push that. Otherwise, default to pushing the artifact in the docker daemon
 	if _, err := layout.FromPath(productTaskOutputInfo.ProductDockerOCIDistOutputDir(dockerID)); err == nil {
-		return runOCIPush(productID, dockerID, productTaskOutputInfo, dryRun, stdout)
+		return runOCIPush(productID, dockerID, productTaskOutputInfo, dryRun, insecure, stdout)
 	}
 	return runDockerDaemonPush(productID, dockerID, productTaskOutputInfo, dryRun, stdout)
 }
 
-func runOCIPush(productID distgo.ProductID, dockerID distgo.DockerID, productTaskOutputInfo distgo.ProductTaskOutputInfo, dryRun bool, stdout io.Writer) error {
+func runOCIPush(productID distgo.ProductID, dockerID distgo.DockerID, productTaskOutputInfo distgo.ProductTaskOutputInfo, dryRun bool, insecure bool, stdout io.Writer) error {
 	index, err := layout.ImageIndexFromPath(productTaskOutputInfo.ProductDockerOCIDistOutputDir(dockerID))
 	if err != nil {
 		return errors.Wrapf(err, "failed to construct image index from OCI layout at path %s", productTaskOutputInfo.ProductDockerOCIDistOutputDir(dockerID))
 	}
 
 	for _, tag := range productTaskOutputInfo.Product.DockerOutputInfos.DockerBuilderOutputInfos[dockerID].RenderedTags {
-		ref, err := name.ParseReference(tag)
+		var ref name.Reference
+		var err error
+		if insecure {
+			ref, err = name.ParseReference(tag, name.Insecure)
+		} else {
+			ref, err = name.ParseReference(tag)
+		}
```

Comment:
I think the following construction is simpler/more straightforward:

```suggestion
		var opts []name.Option
		if insecure {
			opts = append(opts, name.Insecure)
		}
		ref, err := name.ParseReference(tag, opts...)
```